### PR TITLE
fix: crash getItems

### PR DESCRIPTION
**Pull request by luanluciano93**
image

### CODE REVIEW

**Comment by bateunatrave** [comment]
Now the trade window is appearing, but when the other player's item appears, the crash occurs.

**Comment by luanluciano93** [comment]
> Now the trade window is appearing, but when the other player's item appears, the crash occurs.


Can you use the deputy to see where the crash is occurring, like the PR image?

**Comment by kokekanon** [comment]
test in 13.40

in my test there is no crash, but when I accept the trade

> i don't know if the error is in canary or otc. but for some reason
> server sent 0x89 to client.

```lua
ERROR: ProtocolGame parse message exception (357 bytes, 348 unread, last opcode is 0x89 (137), prev opcode is 0xffffffff (-1)): unhandled opcode 137
Packet has been saved to packet.log, you can use it to find what was wrong. (Protocol: 1340)
```  
```lua
src/client/protocolcodes.h
GameServerCloseForgeWindow = 137,


src/client/protocolgame.h
void parseCloseForgeWindow(const InputMessagePtr& msg);


src/client/protocolgameparse.cpp
....
                case Proto::GameServerCloseForgeWindow:
                    parseCloseForgeWindow(msg);
                    break;
                    
...
                    
void ProtocolGame::parseCloseForgeWindow(const InputMessagePtr& msg)
{
    g_lua.callGlobalField("g_game", "onCloseForgeWindow"); // TODO: implement 
}
```
in 8.6 this error does not occur

File: src/client/protocolgameparse.cpp [hunk 1]
```diff
@@ -1498,7 +1498,7 @@ void ProtocolGame::parseOwnTrade(const InputMessagePtr& msg)
     const auto& name = g_game.formatCreatureName(msg->getString());
 
     const uint8_t count = msg->getU8();
-    std::vector<ItemPtr> items;
+    std::vector<ItemPtr> items(count);
```
Comment (nekiro) [inline src/client/protocolgameparse.cpp]: `items.reserve(count)`

File: src/client/protocolgameparse.cpp [hunk 2]
```diff
@@ -1498,7 +1498,7 @@ void ProtocolGame::parseOwnTrade(const InputMessagePtr& msg)
...
 
     for (auto i = 0; i < count; i++) {
         items[i] = getItem(msg);
```
Comment (nekiro) [inline src/client/protocolgameparse.cpp]: `items.push_back(getItem(msg));`

**Review by nekiro — CHANGES_REQUESTED** [review]

**Review by nekiro — APPROVED** [review]

**Pull request merged**